packages/test-lib/test/lib-test.js

```diff
@@ -0,0 +1,17 @@
+const { module, test } = QUnit;
+import { getPuppies, getRats } from "../lib.js";
+
+module("lib test bed", function (hooks) {
+  test("entrypoint exports are exports of the bundle", async function (assert) {
+    // Since this JS is a dep of an HTMLEntrypoint (because it's part of the
+    // test.html's deps) importing the all bundle's exports doesn't necessarily
+    // prove that the JSEntrypoint logic is working (where all exports of the js
+    // entrypoint module are also exports of the bundle). In this case we are
+    // importing 2 of the 3 exports. By visual inspection of the built bundle we
+    // can see that all the exports are indeed included in the bundle--but I'm
+    // unclear how to actually test that and be sure that we aren't actually
+    // seeing HTML Entrypoint logic being tested.
```

Comment:
@ef4 this is looking pretty good. the node build indeed tests from the bundle. However, there is still a quirk, which I outlined in my comment. our entrypoint.json for this has both a js entrypoint (for the js lib), and an html entrypoint (for the test). I want to be sure that I'm not actually testing the HTMLEntrypoint specific logic--and a key difference between these 2 types of entrypoints are the bundle exports. I purposefully did not import all the available exports from the entrypoint so that I could see the JSEntrypoint specific logic where we preserve on the bundle all the entrypoint's exports. I'm unsure how to assert that is the case without inadvertantly seeing the HTMLEntrypoint logic, where we ensure that the bundle exports all the dependencies that are imported from outside of the bundle (which would actually be this test--since it is assigned to a different bundle because of the HTMLEntrypoint).

Reply:
I just added a unit test to explicitly test the exports for js entrypoints in entrypoints.json